### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by almaslennikov**
Update go to 1.20
Update golangci-lint
Update github actions (static checks, arm64 build)

### CODE REVIEW

**Review by e0ne — APPROVED** [review]
LGTM

File: pkg/controllers/controller_suite_test.go [hunk 1]
```diff
@@ -6,7 +6,8 @@ import (
 
 	. "github.com/onsi/ginkgo/v2"
 	. "github.com/onsi/gomega"
-	"k8s.io/klog/v2"
+
+	klog "k8s.io/klog/v2"
```
Comment (ykulazhenkov) [inline pkg/controllers/controller_suite_test.go:10]: Why these aliases are required? Is this something from the linter?

File: Makefile [hunk 2]
```diff
@@ -161,3 +173,11 @@ echo "Downloading $(2)" ;\
...
 }
 endef
+
+define wget-install-tool
+@[ -f $(1) ] || { \
+echo "Downloading $(2)" ;\
+wget -O $(1) $(2);\
+chmod +x $(1) ;\
+}
+endef
```
Comment (ykulazhenkov) [inline Makefile:183]: no new line at the end

File: Makefile [hunk 3]
```diff
@@ -147,6 +155,10 @@ gocovmerge: ## Download gocovmerge if necessary
 gcov2lcov: ## Download gcov2lcov if necessary
 	$(call go-install-tool,$(GCOV2LCOV),github.com/jandelgado/gcov2lcov@v1.0.5)
 
+.PHONY: hadolint
+hadolint: $(BASE) $(HADOLINT_TOOL); $(info  running hadolint...) @ ## Run hadolint
```
Comment (ykulazhenkov) [inline Makefile:159]: please, add target for shellcheck

File: .github/workflows/static-scan.yaml [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+    steps:
+      - name: set up Go
+        uses: actions/setup-go@v3
+        with:
+          go-version: 1.20.x
+      - name: checkout PR
+        uses: actions/checkout@v2
+      - name: run make lint
+        run: make lint
+  shellcheck:
```
Comment (ykulazhenkov) [inline .github/workflows/static-scan.yaml:16]: please, switch to Makefile targets for shellcheck and hadolint

File: .github/workflows/push-release.yml [hunk 5]
```diff
@@ -74,14 +78,5 @@ jobs:
...
-          docker pull ${{ env.REPOSITORY }}:stable-amd64
-          # Create and update manifests
-          docker manifest create ${{ steps.docker_meta.outputs.tags }} ${{ steps.docker_meta.outputs.tags }}-amd64
-          docker manifest annotate ${{ steps.docker_meta.outputs.tags }} ${{ steps.docker_meta.outputs.tags }}-amd64 --arch amd64
-          docker manifest create ${{ env.REPOSITORY }}:stable ${{ env.REPOSITORY }}:stable-amd64
-          docker manifest annotate ${{ env.REPOSITORY }}:stable ${{ env.REPOSITORY }}:stable-amd64 --arch amd64
-          # push manifests
-          docker manifest push ${{ steps.docker_meta.outputs.tags }}
-          docker manifest push ${{ env.REPOSITORY }}:stable
+          docker buildx imagetools create -t ${{ env.REPOSITORY }}:stable ${{ env.REPOSITORY }}:stable-amd64
```
Comment (ykulazhenkov) [inline .github/workflows/push-release.yml:81]: it is possible to specify multiple target tags by providing multiple -t arguments. 
I think it is possible to push required tags with single docker buildx call
```
docker buildx imagetools create -t ${{ env.REPOSITORY }}:stable \
-t ${{ steps.docker_meta.outputs.tags }} \
${{ env.REPOSITORY }}:stable-amd64
```

File: pkg/controllers/controller_suite_test.go (hunk 1 above)
Comment (almaslennikov) [reply pkg/controllers/controller_suite_test.go:10]: Yep, required by linter

File: Makefile (hunk 3 above)
Comment (almaslennikov) [reply Makefile:159]: Actually, we don't have any shell scripts in the repo, so I dropped the shellcheck scan altogether

**Comment by almaslennikov** [comment]
@ykulazhenkov we need to allow github CI actions write access, otherwise hadolint check won't run

File: .github/workflows/static-scan.yaml (hunk 4 above)
Comment (almaslennikov) [reply .github/workflows/static-scan.yaml]: Did so with hadolint, dropped the shellcheck as we don't have any shell scripts in this repo

**Comment by ykulazhenkov** [comment]
> @ykulazhenkov we need to allow github CI actions write access, otherwise hadolint check won't run

current CI issue is not related to actions permissions
```
Downloading https://github.com/hadolint/hadolint/releases/download/v2.12.1-beta/hadolint-Linux-x86_64
/hadolint: Permission denied
make: *** [Makefile:81: /hadolint] Error 1
Error: Process completed with exit code 2.
```

You have bug in the Makefile `HADOLINT_TOOL = $(BINDIR)/hadolint` should be `HADOLINT_TOOL = $(BIN_DIR)/hadolint`

**Review by ykulazhenkov — APPROVED** [review]

**Pull request merged**